modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/DslElementBasedMetadataCacheIdGenerator.java

```diff
@@ -158,9 +162,12 @@ private String getSourceElementName(DslElementModel<?> elementModel) {
 
   private Optional<MetadataCacheId> resolveGlobalElement(DslElementModel<?> elementModel) {
     List<MetadataCacheId> parts = new ArrayList<>();
+    List<String> parameterNamesRequiredForMetadataCacheId =
+        parameterNamesRequiredForMetadataCacheId(elementModel.getModel()).orElse(null);
```

Comment:
you can create the predicate inline here to do the filtering and avoid the null list ( ❌)